stackslib/src/chainstate/nakamoto/staging_blocks.rs

```diff
@@ -188,6 +232,25 @@ impl<'a> NakamotoStagingBlocksConnRef<'a> {
         )))
     }
 
+    /// Get the size of a Nakamoto block, given its index block hash
+    /// Returns Ok(Some(size)) if the block was present
+    /// Returns Ok(None) if there was no such block
+    /// Returns Err(..) on DB error, including block corruption
+    pub fn get_nakamoto_block_size(
+        &self,
+        index_block_hash: &StacksBlockId,
+    ) -> Result<Option<u64>, ChainstateError> {
+        let qry = "SELECT length(data) FROM nakamoto_staging_blocks WHERE index_block_hash = ?1";
+        let args: &[&dyn ToSql] = &[index_block_hash];
+        let res: Option<i64> = query_row(self, qry, args)?;
+        let Some(size_i64) = res else {
+            return Ok(None);
+        };
+        Ok(Some(
+            u64::try_from(size_i64).expect("FATAL: block exceeds i64::MAX"),
+        ))
```

Comment:
Would be a bit simpler with a `.map()`

```suggestion
        let res = query_row(self, qry, args)?
            .map(|size: i64| u64::try_from(size).expect("FATAL: block exceeds i64::MAX"));
        Ok(res)
```